File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.js
function solution(food) {
    const answer = [];
    const [water, ...realFood] = food;
    
    realFood.reverse().forEach((val, idx) => {
        for (let i = 0; i < Math.floor(val / 2); i++) {
            answer.unshift(realFood.length - idx);
            answer.push(realFood.length - idx);
        }
    });
    
    answer.splice(answer.length / 2, 0, 0);
    
    return answer.join('');
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.py
def solution(food):
    water, *real_food = food
    answer = [0]
    
    for idx, val in enumerate(reversed(real_food)):
        count = val // 2
        value = len(real_food) - idx
        answer = [value] * count + answer + [value] * count
    
    return ''.join(map(str, answer))
